standalone/remove-x-status-codes.js

```diff
@@ -0,0 +1,62 @@
+// Remove 404s from request history
+if (typeof println == 'undefined') this.println = print;
+
+// Logging with the script name is super helpful!
+function logger() {
+	print('[' + this['zap.script.name'] + '] ' + arguments[0]);
+}
+
+var ExtensionHistory = Java.type('org.parosproxy.paros.extension.history.ExtensionHistory');
+var Control          = Java.type('org.parosproxy.paros.control.Control');
+var Model            = Java.type('org.parosproxy.paros.model.Model');
+
+var History  = Control.getSingleton().getExtensionLoader().getExtension(ExtensionHistory.class)
+var tree     = Model.getSingleton().getSession().getSiteTree()
+var rootNode = tree.getRoot();	
+
+function hasBadStatusCode(item) {
+  if (item === null) {return true;}
+  return item.getStatusCode() === 404 || item.getStatusCode() === 502 || item.getStatusCode() === null || item.getStatusCode() === 0;
```

Comment:
It should not consider 0 status code as bad, that's for temporary Sites nodes, which should not be removed while in use.